specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-08-01/settings.json

```diff
@@ -0,0 +1,316 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeSettingParameter"
+          },
+          {
+            "$ref": "#/parameters/settingTypeParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Setting deleted successfully."
+          },
+          "204": {
+            "description": "Setting is already deleted or not found."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Setting": {
+      "description": "Setting definition.",
+      "discriminator": "kind",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "Specifies the kind of settings.",
+          "enum": [
+            "taginheritance"
+          ],
+          "x-ms-enum": {
+            "name": "SettingsKind",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "kind"
+      ]
+    },
+    "TagInheritanceSetting": {
+      "description": "Tag Inheritance Setting definition.",
+      "type": "object",
+      "x-ms-discriminator-value": "taginheritance",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Setting"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/TagInheritanceProperties"
+        }
+      }
+    },
+    "SettingsListResult": {
+      "description": "Setting list result. It contains a list of settings.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of settings.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Setting"
+          },
+          "x-ms-identifiers": [
+            "id"
+          ]
+        }
+      }
+    },
+    "TagInheritanceProperties": {
+      "type": "object",
+      "description": "The properties of the tag inheritance setting.",
+      "required": [
+        "preferContainerTags"
+      ],
+      "properties": {
+        "preferContainerTags": {
+          "description": "When resource has the same tag as subscription or resource group and this property is set to true - the subscription or resource group tag will be applied. If subscription and resource group tags are also the same, subscription tag will be applied.",
+          "type": "boolean"
+        }
```

Comment:
Only 3rd option I can imagine is one that works similarly but in reverse - resourceGroups precede subscriptions precedes the resource itself... probably you'll never want it right?

Reply:
But then again, maybe its worth switching it to an enum just in case, or maybe there will be yet other ways to do it in the future